fem/gslib.cpp

```diff
@@ -516,8 +470,347 @@ void FindPointsGSLIB::GetSimplexNodalCoordinates()
+   const Geometry::Type gt = fe->GetGeomType();
+   int NEsplit = 0;
+
+   gsl_mfem_ref -= -1.;  // map  [-1, 1] to
+   gsl_mfem_ref *= 0.5;  //      [0,  1]
+   if (gt == Geometry::SQUARE || gt == Geometry::CUBE) { return; }
+
+   H1_FECollection feclin(1, dim);
+   FiniteElementSpace nodal_fes_lin(meshsplit, &feclin, dim);
+   GridFunction gf_lin(&nodal_fes_lin);
+
+   if (gt == Geometry::TRIANGLE)
+   {
+      const double quad_v[7][2] =
+      {
+         {0, 0}, {0.5, 0}, {1, 0}, {0, 0.5},
+         {1./3., 1./3.}, {0.5, 0.5}, {0, 1}
+      };
+      for (int k = 0; k < dim; k++)
+      {
+         for (int j = 0; j < gf_lin.Size()/dim; j++)
+         {
+            gf_lin(j+k*gf_lin.Size()/dim) = quad_v[j][k];
+         }
+      }
+      NEsplit = 3;
+   }
+   else if (gt == Geometry::TETRAHEDRON)
+   {
+      const double hex_v[15][3] =
+      {
+         {0, 0, 0.}, {1, 0., 0.}, {0., 1., 0.}, {0, 0., 1.},
+         {0.5, 0., 0.}, {0.5, 0.5, 0.}, {0., 0.5, 0.},
+         {0., 0., 0.5}, {0.5, 0., 0.5}, {0., 0.5, 0.5},
+         {1./3., 0., 1./3.}, {1./3., 1./3., 1./3.}, {0, 1./3., 1./3.},
+         {1./3., 1./3., 0}, {0.25, 0.25, 0.25}
+      };
+      for (int k = 0; k < dim; k++)
+      {
+         for (int j = 0; j < gf_lin.Size()/dim; j++)
+         {
+            gf_lin(j+k*gf_lin.Size()/dim) = hex_v[j][k];
+         }
+      }
+      NEsplit = 4;
+   }
+   else if (gt == Geometry::PRISM)
+   {
+      const double hex_v[14][3] =
+      {
+         {0, 0, 0}, {0.5, 0, 0}, {1, 0, 0}, {0, 0.5, 0},
+         {1./3., 1./3., 0}, {0.5, 0.5, 0}, {0, 1, 0},
+         {0, 0, 1}, {0.5, 0, 1}, {1, 0, 1}, {0, 0.5, 1},
+         {1./3., 1./3., 1}, {0.5, 0.5, 1}, {0, 1, 1}
+      };
+      for (int k = 0; k < dim; k++)
+      {
+         for (int j = 0; j < gf_lin.Size()/dim; j++)
+         {
+            gf_lin(j+k*gf_lin.Size()/dim) = hex_v[j][k];
+         }
+      }
+      NEsplit = 3;
+   }
+   else
+   {
+      MFEM_ABORT("Element type not currently supported.");
+   }
+
+   // Simplices are split into quads/hexes for GSLIB. For MFEM, we need to
+   // find the original element number and map the rst from micro to macro element.
+   for (int i = 0; i < points_cnt; i++)
+   {
+      int local_elem   = gsl_elem[i]%NEsplit;
+      gsl_mfem_elem[i] = (gsl_elem[i] - local_elem)/NEsplit; // macro element number
+
+      IntegrationPoint ip;
+      Vector mfem_ref(gsl_mfem_ref.GetData()+i*dim, dim);
+      ip.Set3(mfem_ref.GetData());
+      gf_lin.GetVectorValue(local_elem, ip, mfem_ref); // map to rst of macro element
+   }
+}
+
+void FindPointsGSLIB::Interpolate(const GridFunction &field_in,
+                                  Vector &field_out)
+{
+   const char *gf_name   = field_in.FESpace()->FEColl()->Name();
+   const int  gf_order   = field_in.FESpace()->GetFE(0)->GetOrder(),
+              mesh_order = mesh->GetNodalFESpace()->GetFE(0)->GetOrder();
+
+   if ( strncmp(gf_name, "H1", 2) == 0 && gf_order == mesh_order )
```

Comment:
I think it would be safer to go with `dynamic_cast` for the specific collections that are supported, like `H1_FECollection` etc, and abort for all other cases. Currently, this string comparison would catch collections like H1 Trace, H1 Serendipity, H1 Bernstein, which won't be handled correctly.

Reply:
Yes, I like this idea. Thanks!